via/static/scripts/video_player/hooks/use-app-layout.ts

```diff
@@ -0,0 +1,54 @@
+import type { RefObject } from 'preact';
+import { useCallback, useLayoutEffect, useState } from 'preact/hooks';
+
+/* Relative application container size */
+export type AppSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl';
+
+/**
+ * Return a string representing the relative size of `appContainer` that can
+ * be used to make decisions about app layout and sizing of components. This
+ * would typically be something handled by media queries, but it is an element
+ * (`body`) width that needs to be queried, not the viewport. We could replace
+ * this with container queries once browser support is sufficient.
+ *
+ * NB: This hook will not update if `appContainer.current` changes.
+ */
+export function useAppLayout(
+  appContainer: RefObject<HTMLDivElement | null>
+): AppSize {
+  const [appSize, setAppSize] = useState<AppSize>('sm');
+
+  const updateWidth = useCallback(() => {
+    const containerWidth = appContainer.current?.clientWidth;
+    if (!containerWidth) {
+      return;
+    }
+
+    let containerSize: AppSize = 'sm';
+    if (containerWidth > 1376) {
+      containerSize = '2xl';
+    } else if (containerWidth > 1024) {
+      containerSize = 'xl';
+    } else if (containerWidth > 875) {
+      containerSize = 'lg';
+    } else if (containerWidth > 768) {
+      containerSize = 'md';
+    }
+
+    setAppSize(containerSize);
+  }, [appContainer]);
+
+  useLayoutEffect(() => {
+    const element = appContainer.current;
+    /* istanbul ignore if */
+    if (!element) {
+      return () => {};
+    }
```

Comment:
This state should never occur. Should I remove this guard?

Reply:
I would probably leave as is, or maybe log a warning. You could make an argument that having `appContainer.current` be null is likely a mistake and so the code should fail loudly. On the other hand, the hook is slightly more flexible in its current state.